SYCL/Complex/sycl_complex_math_test_cases.hpp

```diff
@@ -0,0 +1,503 @@
+using std::pair;
+using std::vector;
+
+// Forward decleration of tests
+
+template <typename T> class test_acos;
+template <typename T> class test_asin;
+template <typename T> class test_atan;
+template <typename T> class test_acosh;
+template <typename T> class test_asinh;
+template <typename T> class test_atanh;
+template <typename T> class test_conj;
+template <typename T> class test_cos;
+template <typename T> class test_cosh;
+template <typename T> class test_exp;
+template <typename T> class test_log;
+template <typename T> class test_log10;
+template <typename T> class test_proj;
+template <typename T> class test_sin;
+template <typename T> class test_sinh;
+template <typename T> class test_sqrt;
+template <typename T> class test_tan;
+template <typename T> class test_tanh;
+
+template <typename T> class test_abs;
+template <typename T> class test_arg;
+template <typename T> class test_norm;
+
+template <typename T> class test_polar;
+
+// Stores test cases for each math function used in sycl_complex_math_test.cpp
+// Values are stored in the highest precision type, in this case that is double
+
+template <template <typename> typename test_struct> struct test_cases {
+  static vector<cmplx<double>> std_test_values;
+
+  static const char *test_name;
+
+  bool operator()(sycl::queue &Q) {
+    bool test_passes = true;
+
+    for (auto &test_value : std_test_values) {
+      test_passes &= test_valid_types<test_struct>(Q, test_value);
+    }
+
+    if (!test_passes)
+      std::cerr << test_name << " failed\n";
+
+    return test_passes;
+  }
+};
+
+// test_acos
+template <>
+vector<cmplx<double>> test_cases<test_acos>::std_test_values = {
+    cmplx(1, 1),
+    cmplx(-1, 1),
+    cmplx(1, -1),
+    cmplx(-1, -1),
+    cmplx(INFINITYd, 2.02),
+    cmplx(4.42, INFINITYd),
+    cmplx(INFINITYd, INFINITYd),
+    cmplx(NANd, 2.02),
+    cmplx(4.42, NANd),
+    cmplx(NANd, NANd),
+    cmplx(NANd, INFINITYd),
+    cmplx(INFINITYd, NANd),
+    cmplx(NANd, INFINITYd),
+    cmplx(INFINITYd, NANd),
+};
```

Comment:
This seems to be the common specialization. Could it be made the default value and only have the ones that need different values?

Reply:
Yes, I have made these common specializations.